src/library/scala/concurrent/impl/Promise.scala

```diff
@@ -178,7 +178,9 @@ private[concurrent] object Promise {
    * DefaultPromises, and `linkedRootOf` is currently only designed to be called
    * by Future.flatMap.
    */
-  final class DefaultPromise[T] extends AtomicReference[AnyRef](Nil) with Promise[T] {
+  // Left non-final to enable addition of extra fields by Java/Scala converters
+  // in scala-java8-compat.
+  class DefaultPromise[T] extends AtomicReference[AnyRef](Nil) with Promise[T] {
```

Comment:
> I'm more worried about Java users here.

Are you suggesting that I add final overriding methods that delegate to the super method for each of:

![image](https://cloud.githubusercontent.com/assets/65551/9265431/69fa2180-4276-11e5-98dd-a71984d62027.png)

That seems quite heavyweight to me. What are we seeking to prevent?